removefiles/activity.go

```diff
@@ -84,15 +85,32 @@ func removeByNames(dir, names string) (int, error) {
 		return 0, fmt.Errorf("remove: path: %q: not a directory", dir)
 	}
 
-	files := parseNames(names)
+	var res []*regexp.Regexp
+	for _, p := range split(patterns) {
```

Comment:
I don't like calling this function `split()` because it's too easy to confuse it with `strings.Split()` which has a different signature and result.  I think `splitAndTrim()` or some other name would be better.

What about regex patterns that include a comma?  I think for regex we are better off using a linebreak as a separator, because it's not used in the regex syntax.  Unfortunately the default `viper.Unmarshal()`) decode hooks are going to get in our way here, because by default it splits string configs on the comma into a slice.  We'll have to override the default viper decode hooks like I did here: https://github.com/artefactual-sdps/enduro/pull/939/commits/a5e481e941114db4b10c0e35a26f6b32ddd8efca#diff-54c7c1af5fa8d5db4dc49f0e8e80e93ba2b1183ba4d5c9e2e5729e6deae6a3cdR96 to avoid the `viper.Unmarshal()` string splitting behaviour.

I would also just prefer the function to have a second "delimiter" param to match the `strings.Split()` signature, and it would allow us to use a different delimiter for the names and patterns list.